openapi_python_client/templates/model.pyi

```diff
@@ -105,27 +106,69 @@ class {{ model.reference.class_name }}:
             {{ construct(model.additional_properties, "prop_dict") | indent(12) }}
             additional_properties[prop_name] = {{ model.additional_properties.python_name }}
 
-        {{model.reference.module_name}}.additional_properties = additional_properties
+        {{model.reference.module_name}}._additional_properties = additional_properties
     {% else %}
-        {{model.reference.module_name}}.additional_properties = d
+        {{model.reference.module_name}}._additional_properties = d
     {% endif %}
 {% endif %}
         return {{model.reference.module_name}}
 
     {% if model.additional_properties %}
     @property
     def additional_keys(self) -> List[str]:
-        return list(self.additional_properties.keys())
+        return list(self._additional_properties.keys())
 
     def __getitem__(self, key: str) -> {{ additional_property_type }}:
-        return self.additional_properties[key]
+        return self._additional_properties[key]
 
     def __setitem__(self, key: str, value: {{ additional_property_type }}) -> None:
-        self.additional_properties[key] = value
+        self._additional_properties[key] = value
 
     def __delitem__(self, key: str) -> None:
-        del self.additional_properties[key]
+        del self._additional_properties[key]
 
     def __contains__(self, key: str) -> bool:
-        return key in self.additional_properties
+        return key in self._additional_properties
     {% endif %}
+
+{% macro required_property_type(prop) %}
```

Comment:
Ideally, this would go in the code generator itself (e.g., as a method on `prop`). This would reduce duplicated code, and make the type nicer in some cases (e.g., avoiding things like `Optional[None]`). However, given the slowness of incorporating changes upstream, I'm fine with this.